src/main/TreeQuery.cs

```diff
@@ -0,0 +1,209 @@
+
+            Postsynaptic.AppendQuery(typeFromHandle.GetQueryKey("Postsynaptic"), stringBuilder);
+            RegionId.AppendQuery(typeFromHandle.GetQueryKey("RegionId"), stringBuilder);
+
+            // URL encode the AvatarUrl values when converting to string
+            AvatarUrl?.ToList().ForEach(url =>
+            {
+                if (!string.IsNullOrEmpty(url))
+                {
+                    string encodedUrl = HttpUtility.UrlEncode(url);
+                    if (stringBuilder.Length > 0) stringBuilder.Append("&");
+                    stringBuilder.Append($"{typeFromHandle.GetQueryKey("AvatarUrl")}={encodedUrl}");
+                }
+            });
+
+            ExpandUntilPostsynapticMirrors.AppendQuery(typeFromHandle.GetQueryKey("ExpandUntilPostsynapticMirrors"), stringBuilder);
+            DirectionValues.AppendQuery(typeFromHandle.GetQueryKey("DirectionValues"), delegate (DirectionValues v)
+            {
+                int num = (int)v;
+                return num.ToString();
+            }, stringBuilder);
+
+            if (stringBuilder.Length > 0)
+            {
+                stringBuilder.Insert(0, '?');
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        public static bool TryParse(string value, out TreeQuery result)
+        {
+            result = null;
+            bool result2 = false;
+
+            if (!string.IsNullOrWhiteSpace(value))
+            {
+                value = (value.StartsWith("?") ? value.Substring(1) : value);
+
+                // Parse query parameters with proper URL decoding
+                var parameters = ParseQueryString(value);
+
+                // Convert dictionary to string array format for existing extension methods
+                var parameterArray = ConvertParametersToArray(parameters);
+                string[] array = parameterArray.ToArray();
+
+                if (array.Length != 0)
+                {
+                    try
+                    {
+                        Type typeFromHandle = typeof(TreeQuery);
+
+                        result = new TreeQuery
+                        {
+                            Postsynaptic = array.GetQueryArrayOrDefault(typeFromHandle.GetQueryKey("Postsynaptic")),
+                            RegionId = array.GetQueryArrayOrDefault(typeFromHandle.GetQueryKey("RegionId")),
+                            DirectionValues = array.GetNullableEnumValue<DirectionValues>(typeFromHandle.GetQueryKey("DirectionValues")),
+                            AvatarUrl = GetDecodedValues(parameters, typeFromHandle.GetQueryKey("AvatarUrl")),
+                            ExpandUntilPostsynapticMirrors = array.GetQueryArrayOrDefault(typeFromHandle.GetQueryKey("ExpandUntilPostsynapticMirrors"))
+                        };
+
+                        // Parse the AvatarUrl to extract additional query parameters (existing logic)
+                        if (result.AvatarUrl != null && result.AvatarUrl.Any())
```

Comment:
No need to parse the components of the AvatarUrl. The ticket only requested for the 5 properties on lines 89 to 93.